### PR TITLE
add progress bar for unit evalution

### PR DESCRIPTION
**Pull request by adumasphi**

### CODE REVIEW

**Comment by adumasphi** [comment]
cc @jschueller

File: otwrapy/_otwrapy.py [hunk 1]
```diff
@@ -337,6 +339,36 @@ def __exit__(self, type, value, traceback):
...
+    """
+
+    def _exec_sample(X):
+        X = ot.Sample(X)
+        Y = ot.Sample(0, func.getOutputDimension())
+        nb_flush = X.getSize()
+        for i, x in enumerate(X):
+            Y.add(func(x))
+            if verbosity > 0 and X.getSize() > 1:
+                update_progress(i, X.getSize(), 'Sample evaluation', nb_flush, bar_length=50)
```
Comment (jschueller) [inline otwrapy/_otwrapy.py]: maybe just use tqdm ?

Comment (adumasphi) [reply otwrapy/_otwrapy.py]: Modification faite

File: otwrapy/_otwrapy.py [hunk 2]
```diff
@@ -574,7 +609,7 @@ def __init__(self, wrapper, backend='multiprocessing', n_cpus=-1, verbosity=10,
 
         # This configures how to run samples on the model :
         if self.n_cpus == 1:
-            self._exec_sample = self.wrapper
+            self._exec_sample = _exec_sample_unit_progress_bar(self.wrapper, verbosity)
```
Comment (jschueller) [inline otwrapy/_otwrapy.py:609]: lets actually allow a "progress_bar" backend keyword, so the condition rewrites `if backend=="progress_bar" or self.n_cpus == 1`
it should be renamed "serial" though which is more explicit (rename the function _exec_sample_serial too)

File: tests/test_wrapper.py [hunk 3]
```diff
@@ -8,6 +8,9 @@ def backendtest(backend):
     ot.RandomGenerator.SetSeed(0)
     n_cpu = multiprocessing.cpu_count()
     sizes = [1, n_cpu, n_cpu + 1]
+    if backend == 'progress_bar':
+        backend = 'multiprocessing'  # default backend not really used
+        n_cpu = 1
```
Comment (jschueller) [inline tests/test_wrapper.py]: then here we could actually drop this and allow for the "serial" backend

File: README.rst [hunk 4]
```diff
@@ -9,6 +9,7 @@ on top of `OpenTURNS <http://www.openturns.org>`_, with its users as the target
 audience. Documentation is available
 `here <http://openturns.github.io/otwrapy/master>`_. The module provides :
 
+- An integrated progress bar for unit evaluation of a sample without parallelization.
```
Comment (jschueller) [inline README.rst:12]: to remove?

Comment (adumasphi) [reply README.rst:12]: You don't think it is relevant ?

Comment (jschueller) [reply README.rst:12]: oh yes nevermind, I thought you mentioned it because of the old progress bar module

**Pull request merged**